### PR TITLE
Add ML.NET notes from session 5

### PR DESCRIPTION
**Pull request by terrajobst**

### CODE REVIEW

File: 2018/Microsoft.ML.Core/Session5.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            options.HasHeader = true,
+            options.Column.Add("SepalLength", DataKind.R4, 0);
+            options.Column.Add("SepalWidth", DataKind.R4, 1);
+            options.Column.Add("PetalLength", DataKind.R4, 2);
+            options.Column.Add("PetalWidth", DataKind.R4, 3);
+            options.Column.Add("Label", DataKind.Text, 4);
+
+            var reader = mlContext.Data.CreateTextFileSource(options);
+            ```
+    - Tom check what you were plan on checking :-)
```
Comment (shauheen) [inline 2018/Microsoft.ML.Core/Session5.md]: 💯

File: 2018/Microsoft.ML.Core/Session5.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+Status: **Needs more work**
+
+## Notes
+
+* `MLContext`
+    - Solves two problems: the chaining for `IHostEnvironment` and the
+      discoverability for finding implementations of common abstraction.
+    - They are solved by offering extension methods
+    - We should align the naming of `XxxCatalog` with `XxxContext`
```
Comment (TomFinley) [inline 2018/Microsoft.ML.Core/Session5.md]: [Issue here.](https://github.com/dotnet/machinelearning/issues/1796)

File: 2018/Microsoft.ML.Core/Session5.md (hunk 1 above)
Comment (TomFinley) [inline 2018/Microsoft.ML.Core/Session5.md]: It turns out the answer was, "we did both." 😛 [Issue here.](https://github.com/dotnet/machinelearning/issues/1797)

File: 2018/Microsoft.ML.Core/Session5.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      but sentinel values seem more problem. Overloads don't work very well
+      because the types are the same and it will force people to specify value
+      of the other. Easiest solution might be to make both nullable. Still not
+      idiomatic API design, but at least consistent.
+* [Typical example](https://dotnet.microsoft.com/learn/machinelearning-ai/ml-dotnet-get-started-tutorial)
+    - Most of the methods aren't verbs; instead of `TextReader()` it should be
+      `CreateTextReader()`
+    - We should have a pattern for the arguments:
+        - Simple set of arguments should just be parameters
+        - More complicated arguments should be an option class that is mutable
```
Comment (TomFinley) [inline 2018/Microsoft.ML.Core/Session5.md]: Entered as [this issue](https://github.com/dotnet/machinelearning/issues/1798), this was a fairly lengthy discussion that went into more things than are captured in this paragraph so it's possible I forgot something in the issue, let me know if so.

File: 2018/Microsoft.ML.Core/Session5.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    - It's a bit odd that the constructor uses nullability to indicate default,
+      but sentinel values seem more problem. Overloads don't work very well
+      because the types are the same and it will force people to specify value
+      of the other. Easiest solution might be to make both nullable. Still not
+      idiomatic API design, but at least consistent.
+* [Typical example](https://dotnet.microsoft.com/learn/machinelearning-ai/ml-dotnet-get-started-tutorial)
+    - Most of the methods aren't verbs; instead of `TextReader()` it should be
+      `CreateTextReader()`
+    - We should have a pattern for the arguments:
+        - Simple set of arguments should just be parameters
```
Comment (TomFinley) [inline 2018/Microsoft.ML.Core/Session5.md]: This is captured in [existing issues](https://github.com/dotnet/machinelearning/issues/1690) but to reinforce also mentioned it as a subitem [in a new issue as well](https://github.com/dotnet/machinelearning/issues/1798).

**Pull request merged**